platforms/access-bridge/src/services/stripe-service.ts

```diff
@@ -57,4 +60,40 @@ export class StripeService {
 
     return productsWithPrices;
   }
+
+  /**
+   * Creates a Stripe Checkout session based on the provided price ID.
+   * @param viewer Email address and viewer id from the auth token used for creating the checkout session.
+   * @param params Stripe checkout params to use for creating the checkout session.
+   * @returns A Promise resolving to a Stripe Checkout Session object, including a URL for the checkout page.
+   */
+  async createCheckoutSession(viewer: Viewer, params: StripeCheckoutParams): Promise<Stripe.Checkout.Session> {
+    const sessionParams: Stripe.Checkout.SessionCreateParams = {
+      payment_method_types: ['card'],
+      line_items: [
+        {
+          price: params.price_id,
+          quantity: 1,
+        },
+      ],
+      metadata: {
+        viewer_id: viewer.id,
+      },
+      customer_email: viewer.email,
+      mode: params.mode,
+      success_url: params.success_url,
+      cancel_url: params.cancel_url,
+
+      // Conditionally include `subscription_data` only if mode is `subscription`
+      ...(params.mode === 'subscription' && {
+        subscription_data: {
+          metadata: {
+            viewer_id: viewer.id,
```

Comment:
Please add a comment here for how important this is to the current implementation (it's our only way of connecting the payment back to this user.) I really want to revisit this approach in Q4, because this does not give me a warm fuzzy sense of reliability. Do you know why we aren't creating customers in stripe or using the reference id?
https://docs.stripe.com/api/customers/object
https://docs.stripe.com/api/checkout/sessions/object#checkout_session_object-client_reference_id

Reply:
Stripe creates a customer out of the box when a payment is made. Sims team already tested and used this stuff and no complains were made about something missing. We can always add some stuff if needed.